### PR TITLE
Adding OneStopEnglish corpus dataset

### PR DESCRIPTION
**Pull request by purvimisal**
This PR adds OneStopEnglish Corpus containing texts classified into reading levels (elementary, intermediate, advance) for automatic readability assessment and text simplification. 

Link to the paper: https://www.aclweb.org/anthology/W18-0535.pdf

### CODE REVIEW

File: datasets/onestop_english/onestop_english.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+        data_dir = os.path.join(data_dir, "Texts-SeparatedByReadingLevel")
+
+        ele_samples = []
+        dir_path = data_dir + "/Ele-Txt/"
+        files = os.listdir(dir_path)
+        print("DIRPATH", dir_path)
+        for f in files:
+            try:
+                with open(dir_path + f, encoding="utf-8-sig") as myfile:
```
Comment (lhoestq) [inline datasets/onestop_english/onestop_english.py:100]: can you use os.path.join here ?

File: datasets/onestop_english/onestop_english.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            except Exception as e:
+                print("Error with:", dir_path + f, e)
+
+        int_samples = []
+        dir_path = data_dir + "/Int-Txt/"
+        print("DIRPATH", dir_path)
+        files = os.listdir(dir_path)
+        for f in files:
+            try:
+                with open(dir_path + f, encoding="utf-8-sig") as myfile:
```
Comment (lhoestq) [inline datasets/onestop_english/onestop_english.py:112]: same here

File: datasets/onestop_english/onestop_english.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            except Exception as e:
+                print("Error with:", dir_path + f, e)
+
+        adv_samples = []
+        dir_path = data_dir + "/Adv-Txt/"
+        print("DIRPATH", dir_path)
+        files = os.listdir(dir_path)
+        for f in files:
+            try:
+                with open(dir_path + f, encoding="utf-8-sig") as myfile:
```
Comment (lhoestq) [inline datasets/onestop_english/onestop_english.py:124]: same here

File: datasets/onestop_english/onestop_english.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+_HOMEPAGE = "https://github.com/nishkalavallabhi/OneStopEnglishCorpus"
+
+_LICENSE = "Creative Commons Attribution-ShareAlike 4.0 International License"
+
+_URL = "https://github.com/purvimisal/OneStopCorpus-Compiled/raw/main/Texts-SeparatedByReadingLevel.zip"
+
+
+# TODO: Name of the dataset usually match the script name with CamelCase instead of snake_case
+class OneStopEnglish(datasets.GeneratorBasedBuilder):
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    def _get_examples_from_split(self, split_key, data_dir):
+        """Reads the downloaded and extracted files and combines the individual text files to one dataset and split it into 80% train,
+        10% validation and 10% test data."""
+
+        data_dir = os.path.join(data_dir, "Texts-SeparatedByReadingLevel")
+
+        ele_samples = []
+        dir_path = data_dir + "/Ele-Txt/"
+        files = os.listdir(dir_path)
+        print("DIRPATH", dir_path)
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        train_samples = ele_samples[:i1] + int_samples[:i1] + adv_samples[:i1]
+        train_labels = (["ele"] * i1) + (["int"] * i1) + (["adv"] * i1)
+        validation_samples = ele_samples[i1:i2] + int_samples[i1:i2] + adv_samples[i1:i2]
+        validation_labels = (["ele"] * (i2 - i1)) + (["int"] * (i2 - i1)) + (["adv"] * (i1 - i1))
+        test_samples = ele_samples[i2:] + int_samples[i2:] + adv_samples[i2:]
+        test_labels = (
+            (["ele"] * (len(ele_samples) - i2))
+            + (["int"] * (len(ele_samples) - i2))
+            + (["adv"] * (len(ele_samples) - i2))
+        )
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        ]
+
+    def _get_examples_from_split(self, split_key, data_dir):
+        """Reads the downloaded and extracted files and combines the individual text files to one dataset and split it into 80% train,
+        10% validation and 10% test data."""
+
+        data_dir = os.path.join(data_dir, "Texts-SeparatedByReadingLevel")
+
+        ele_samples = []
+        dir_path = data_dir + "/Ele-Txt/"
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        for f in files:
+            try:
+                with open(dir_path + f, encoding="utf-8-sig") as myfile:
+                    text = myfile.read().strip()
+                    ele_samples.append(text)
+            except Exception as e:
+                print("Error with:", dir_path + f, e)
+
+        int_samples = []
+        dir_path = data_dir + "/Int-Txt/"
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+        for f in files:
+            try:
+                with open(dir_path + f, encoding="utf-8-sig") as myfile:
+                    text = myfile.read().strip()
+                    int_samples.append(text)
+            except Exception as e:
+                print("Error with:", dir_path + f, e)
+
+        adv_samples = []
+        dir_path = data_dir + "/Adv-Txt/"
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+        print("DIRPATH", dir_path)
+        files = os.listdir(dir_path)
+        for f in files:
```
[the remark this answers is not in the document]
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py (hunk 1 above)
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py (hunk 2 above)
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

File: datasets/onestop_english/onestop_english.py (hunk 3 above)
Comment (purvimisal) [reply datasets/onestop_english/onestop_english.py]: Done

**Comment by purvimisal** [comment]
Hi @lhoestq, thanks for the review.
I have made all the changes, PTAL! :)

**Review by lhoestq — APPROVED** [review]
nice thank you !

**Pull request merged**